### DISCUSSION TITLE
How to map Ulauncher to use env python instead of base python

### DISCUSSION THREAD

**Discussion by behloolsabir** [Ulauncher/Ulauncher#1253] [Q&A]
This app's full potential is achieved only after adding numerous extensions, which requires the installation of many python libraries. Is there a way to specify a particular python environment to mitigate the risk of using an experimental extension that may require the installation of an extension that could potentially damage your base python installation?

**Comment by troycurtisjr** [comment]
Note that the extensions themselves are not installed into the system environment, they are executed from a local ulauncher directory. Of course if any libraries are necessary then yes they currently require installation into the system unless PYTHONPATH is manually override (which could be done for your usage if desired).

Separately I have started on a method to pull in dependency using a virtual environment automatically as part of the quite old discussion in https://github.com/Ulauncher/Ulauncher/issues/273, but I've been a bit distracted by other projects...

**Reply by behloolsabir** [reply]
Thank you for your response. Can you provide more information on where to change the PYTHONPATH? Additionally, is it necessary to perform a fresh install of Ulauncher to achieve this outcome?

What I tried: 
* in an already installed Ulauncher I changed [this](https://github.com/Ulauncher/Ulauncher/blob/609aa9bc60b2163567102c26e83e84951414a738/bin/ulauncher#L13) to `python_path = ['/mnt/env0/bin/python']` which is the python path I wish to use. 
* Ulauncher still complains about `/usr/bin/python3: No module named pip`.

**Reply by friday** [reply]
Don't change the python path @behloolsabir.  Maybe you can change the "hash bang" in the ulauncher binary (first line of `cat $(which ulauncher)`), but imo this is still a bad idea, and a scenario we don't support.

**Comment by kskollepara** [comment]
This is still a pertinent question. Hoping to see a solution to make full use of ulauncher

**Reply by friday** [reply]
See #273 and #721
